.github/workflows/build-and-test.yml

```diff
@@ -0,0 +1,21 @@
+name: Xcode - Build, Analyze, & Test
```

Comment:
Is there a benefit to having this run in addition to the existing circleci configuration?

Reply:
I added it just to get a feel for how easy or hard it is to do it in GH Actions instead of CircleCI. _My_ preference is to do it in GH, if for no other reason than keeping the configuration in source control. I've also noticed that when I added Swift code to the SDK on a branch, the GH action succeeded, but CircleCI failed. 🤷

Reply:
> _My_ preference is to do it in GH

+1, if we can get this working well in GH, would be a nice simplification to run everything in GH Actions.

Reply:
Might also be a good time to add a status badge for the master branch https://docs.github.com/en/actions/monitoring-and-troubleshooting-workflows/adding-a-workflow-status-badge

Reply:
Not a bad idea, but let's do it after this is merged.